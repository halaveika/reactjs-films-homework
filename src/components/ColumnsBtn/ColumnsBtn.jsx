import React from 'react';
import PropTypes from 'prop-types';
import './ColumnsBtn.scss';

export default function ColumnsBtn({setListFlexDirection}) {
  return (
 <div className="columns-btn" onClick={()=>setListFlexDirection(false)}>
   <div></div>
   <div></div>
 </div>
  );
}

ColumnsBtn.propTypes = {
  setListFlexDirection: PropTypes.func.isRequired,
};
