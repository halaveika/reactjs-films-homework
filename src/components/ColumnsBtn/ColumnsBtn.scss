.columns-btn,
.columns-btn_active {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: space-around;
  width: 20px;
  height: 20px;
  cursor: pointer;

  & div {
    width: 100%;
    height: 8px;
    background-color: #ced1d4;
  }

  &_active div {
    background-color: #1890ff;
  }

  &:hover div {
    background-color: #1890ff;
  }
}
