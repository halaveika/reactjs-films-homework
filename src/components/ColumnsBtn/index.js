import ColumnsBtn from './ColumnsBtn';

export default ColumnsBtn;
