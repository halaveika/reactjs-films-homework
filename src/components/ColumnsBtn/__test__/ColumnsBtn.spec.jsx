import React from 'react';
import { create, act } from 'react-test-renderer';
import ColumnsBtn from '../ColumnsBtn';

  const setListFlexDirection = jest.fn();


describe('test ColumnsBtn component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const component = create(
      <ColumnsBtn
      setListFlexDirection={setListFlexDirection}
      />,);

  it('should render ColumnsBtn component', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle onClick event',  () => {
    const { root } = component;
    const clickable = root.findByProps({ className: 'columns-btn' });
    act(() => { clickable.props.onClick(); });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(setListFlexDirection).toBeCalled();
  });
});
