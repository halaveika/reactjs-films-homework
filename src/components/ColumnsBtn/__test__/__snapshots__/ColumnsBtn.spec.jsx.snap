// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ColumnsBtn component should handle onClick event 1`] = `
<div
  className="columns-btn"
  onClick={[Function]}
>
  <div />
  <div />
</div>
`;

exports[`test ColumnsBtn component should render ColumnsBtn component 1`] = `
<div
  className="columns-btn"
  onClick={[Function]}
>
  <div />
  <div />
</div>
`;
